src/plugins/external_tool_support/src/blast/align_worker_subtasks/BlastAlignToReferenceTask.cpp

```diff
@@ -0,0 +1,371 @@
+        auto subtask = qobject_cast<BlastAlignToReferenceTask *>(task.data());
+        SAFE_POINT(subtask != nullptr, "BlastAlignToReferenceMuxTask has wrong subtask", ReportResult_Finished);
+        alignmentResults << subtask->getAlignmentResults();
+    }
+    return ReportResult_Finished;
+}
+
+const QList<AlignToReferenceResult> &BlastAlignToReferenceMuxTask::getAlignmentResults() const {
+    return alignmentResults;
+}
+
+/************************************************************************/
+/* BlastAlignToReferenceTask */
+/************************************************************************/
+BlastAlignToReferenceTask::BlastAlignToReferenceTask(const QString &_blastDbPath,
+                                                     const QList<SharedDbiDataHandler> &_reads,
+                                                     const SharedDbiDataHandler &_reference,
+                                                     DbiDataStorage *_storage,
+                                                     const QString &taskNameSuffix)
+    : Task(tr("Align reads with BLAST & Smith-Waterman task") + taskNameSuffix, TaskFlags_NR_FOSE_COSC),
+      dbPath(_blastDbPath),
+      reads(_reads),
+      reference(_reference),
+      storage(_storage) {
+    QScopedPointer<U2SequenceObject> referenceSequence(StorageUtils::getSequenceObject(storage, reference));
+    CHECK_EXT(referenceSequence->getSequenceLength() < INT_MAX, setError(tr("Maximum supported reference sequence length is 2Gb")), );
+    setMaxParallelSubtasks(MAX_PARALLEL_SUBTASKS_AUTO);
+}
+
+void BlastAlignToReferenceTask::prepare() {
+    CHECK(!reads.isEmpty(), );
+
+    QScopedPointer<U2SequenceObject> referenceSequence(StorageUtils::getSequenceObject(storage, reference));
+    CHECK_EXT(!referenceSequence.isNull(), setError(tr("Reference sequence is not found")), );
+
+    BlastTaskSettings settings;
+    settings.programName = "blastn";
+    settings.databaseNameAndPath = dbPath;
+    settings.wordSize = 11;
+    settings.xDropoffGA = 20;
+    settings.xDropoffUnGA = 10;
+    settings.xDropoffFGA = 100;
+    settings.numberOfProcessors = AppContext::getAppSettings()->getAppResourcePool()->getIdealThreadCount();
+    settings.numberOfHits = 1;  // 'culling_limit' - keep the longest hit.
+    settings.gapOpenCost = 2;
+    settings.gapExtendCost = 2;
+    for (const SharedDbiDataHandler &read : qAsConst(reads)) {
+        QScopedPointer<U2SequenceObject> readObject(StorageUtils::getSequenceObject(storage, read));
+        CHECK_EXT(!readObject.isNull(), setError(L10N::nullPointerError("U2SequenceObject")), );
+
+        QByteArray readSequence = readObject->getWholeSequenceData(stateInfo);
+        CHECK_OP(stateInfo, );
+        bool isEmptyReadSequence = std::all_of(readSequence.begin(), readSequence.end(), [](char c) { return c == U2Msa::GAP_CHAR || c == 'N'; });
+        CHECK_EXT(!isEmptyReadSequence, setError(tr("Read doesn't contain meaningful data")), );
```

Comment:
Maybe `CHECK_CONTINUE`? Or, at least, specify the read's name

Reply:
Correct we need to register 'bad read' here in the results. A flow of the old code. Will fix it as a part of this PR